fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/DefaultScheduledDateGenerator.java

```diff
@@ -96,10 +96,33 @@ public LocalDate generateNextRepaymentDate(final LocalDate lastRepaymentDate, fi
                         loanApplicationTerms.getNumberOfdays());
             }
         }
-        
+        if (currentCalendar == null && holidayDetailDTO.getWorkingDays().getExtendTermForRepaymentsOnHolidays()) {
```

Comment:
if extendTermForRepaymentsOnHolidays is true, why can't we call adjustRepaymentDate method  and return that date?